### PR TITLE
STY: Replace .format() to fstrings

### PR DESCRIPTION
**Pull request by EydenVillanueva**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

#29547

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@EydenVillanueva Thanks for the PR. can you merge master to resolve the merge conflicts.

File: pandas/core/missing.py [hunk 1]
```diff
@@ -119,10 +116,7 @@ def clean_interp_method(method, **kwargs):
     if method in ("spline", "polynomial") and order is None:
         raise ValueError("You must specify the order of the spline or polynomial.")
     if method not in valid:
-        raise ValueError(
-            "method must be one of {valid}. Got '{method}' "
-            "instead.".format(valid=valid, method=method)
-        )
+        raise ValueError(f"method must be one of {valid}. Got '{method}' " "instead.")
```
Comment (simonjayhawkins) [inline pandas/core/missing.py]:
```suggestion
        raise ValueError(f"method must be one of {valid}. Got '{method}' instead.")
```

File: pandas/core/indexes/period.py [hunk 2]
```diff
@@ -445,10 +444,11 @@ def __array_wrap__(self, result, context=None):
                     return Index(result, name=name)
             elif isinstance(func, np.ufunc):
                 if "M->M" not in func.types:
-                    msg = "ufunc '{0}' not supported for the PeriodIndex"
                     # This should be TypeError, but TypeError cannot be raised
                     # from here because numpy catches.
-                    raise ValueError(msg.format(func.__name__))
+                    raise ValueError(
+                        f"ufunc {func.__name__}" " not supported for the PeriodIndex"
```
Comment (simonjayhawkins) [inline pandas/core/indexes/period.py]:
```suggestion
                        f"ufunc {func.__name__} not supported for the PeriodIndex"
```

File: pandas/core/missing.py [hunk 3]
```diff
@@ -375,8 +369,7 @@ def _interpolate_scipy_wrapper(
         # GH #10633, #24014
         if isna(order) or (order <= 0):
             raise ValueError(
-                "order needs to be specified and greater than 0; "
-                "got order: {}".format(order)
+                "order needs to be specified and greater than 0; " f"got order: {order}"
```
Comment (simonjayhawkins) [inline pandas/core/missing.py]:
```suggestion
                f"order needs to be specified and greater than 0; got order: {order}"
```

**Comment by EydenVillanueva** [comment]
> @EydenVillanueva Thanks for the PR. can you merge master to resolve the merge conflicts.

Done 👍

File: pandas/core/indexes/accessors.py [hunk 4]
```diff
@@ -26,8 +26,8 @@ class Properties(PandasDelegate, PandasObject, NoNewAttributesMixin):
     def __init__(self, data, orig):
         if not isinstance(data, ABCSeries):
             raise TypeError(
-                f"cannot convert an object of type {type(data)} to a "
-                "datetimelike index"
+                "cannot convert an object of type {0} to a "
+                "datetimelike index".format(type(data))
```
Comment (simonjayhawkins) [inline pandas/core/indexes/accessors.py]: @EydenVillanueva It looks like the merge undid some changes on master. can you merge again, see https://dev.pandas.io/docs/development/contributing.html#updating-your-pull-request

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@EydenVillanueva IIUC there should only be 3 or 4 files changed?

**Comment by EydenVillanueva** [comment]
> @EydenVillanueva IIUC there should only be 3 or 4 files changed?

Yes, you're right. It should be fine now. 
I'll keep checking in case you find another problem

**Review by WillAyd — APPROVED** [review]
lgtm merge if happy @simonjayhawkins

**Review by simonjayhawkins — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @EydenVillanueva

**Pull request merged**